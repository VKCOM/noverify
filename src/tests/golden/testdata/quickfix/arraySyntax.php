<?php

class Foo {}

$_ = array(1,2,3);
$_ = array( new Foo,    "Hello",
        1,2,
        3,4,
);

function f() {
    $_ = array();
    $_ = array(1,2,);
    $_ = array(1,2,3);
    $_ = array(new Foo, new Foo);
    $_ = array("info" => new Foo, new Foo);
    $_ = array("info" => new Foo, "for" => 2, "home" => function() {});

    $_ = array(
        "info" => new Foo,
        new Foo
    );
    $_ = array(
        1,2,
        3,4,
    );
    $_ = array( new Foo,    "Hello",
        1,2,
        3,4,
    );

    b(array(1,2));

    // So far, we cannot replace recursively, so replacement in
    // multidimensional arrays will only be performed for external arrays.
    $_ = array(array(1,2,3), array(1,2,3));
    $_ = array(array("info" => new Foo, "for" => 2, "home" => function() {}), array(1,2,3));
}

function b($a) {
    switch ($a) {
    case 1:
        return array(1, 2, 3);
    case 2:
        return array("info" => new Foo, "for" => 2, "home" => function() {});
    }
}
