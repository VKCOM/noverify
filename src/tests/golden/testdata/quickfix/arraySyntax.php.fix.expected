<?php

class Foo {}

$_ = [1,2,3];
$_ = [ new Foo,    "Hello",
        1,2,
        3,4,
];

function f() {
    $_ = [];
    $_ = [1,2,];
    $_ = [1,2,3];
    $_ = [new Foo, new Foo];
    $_ = ["info" => new Foo, new Foo];
    $_ = ["info" => new Foo, "for" => 2, "home" => function() {}];

    $_ = [
        "info" => new Foo,
        new Foo
    ];
    $_ = [
        1,2,
        3,4,
    ];
    $_ = [ new Foo,    "Hello",
        1,2,
        3,4,
    ];

    b([1,2]);

    // So far, we cannot replace recursively, so replacement in
    // multidimensional arrays will only be performed for external arrays.
    $_ = [array(1,2,3), array(1,2,3)];
    $_ = [array("info" => new Foo, "for" => 2, "home" => function() {}), array(1,2,3)];
}

function b($a) {
    switch ($a) {
    case 1:
        return [1, 2, 3];
    case 2:
        return ["info" => new Foo, "for" => 2, "home" => function() {}];
    }
}
